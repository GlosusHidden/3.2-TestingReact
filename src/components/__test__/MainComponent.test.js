import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import MainComponent from '../MainComponent'

import {myStore, mockStore} from './test_data/data'

it('MainComponent Component Snapshot', () => {
    const geolocationMock = {
      getCurrentPosition: jest.fn()
    };
    global.navigator.geolocation = geolocationMock;
    const renderedValue = renderer.create(
      <Provider store={mockStore}>
        <MainComponent />
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
})
