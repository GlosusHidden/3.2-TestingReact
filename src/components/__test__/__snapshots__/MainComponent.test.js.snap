// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent Component Snapshot 1`] = `
<div
  className="MainComponent"
>
  <div
    id="nav_refresh_geolocation"
  >
    <div
      className="text"
    >
      Weather here
    </div>
    <button
      className="button"
      id="refresh_geolocation"
      onClick={[Function]}
    >
      Refresh Geolocation
    </button>
  </div>
  <div
    className="geolocation_weather"
  >
    <div
      className="main_geo_weather_info"
    >
      <div
        className="city"
      >
        Farforovaya
      </div>
      <div
        className="img"
      >
        <img
          alt="weather"
          className="weather_icon"
          src="http://openweathermap.org/img/wn/04n@2x.png"
        />
      </div>
      <div
        className="temp"
      >
        3
        °C
      </div>
    </div>
    <div
      className="list"
      id="advanced_weather_info"
    >
      <div
        className="line"
      >
        <div
          className="param"
        >
          Weather
        </div>
        <div
          className="value"
        >
          Overcast clouds
        </div>
      </div>
      <div
        className="line"
      >
        <div
          className="param"
        >
          Pressure
        </div>
        <div
          className="value"
        >
          767 mmHg
        </div>
      </div>
      <div
        className="line"
      >
        <div
          className="param"
        >
          Humidity
        </div>
        <div
          className="value"
        >
          90 %
        </div>
      </div>
      <div
        className="line"
      >
        <div
          className="param"
        >
          Wind
        </div>
        <div
          className="value"
        >
          3
        </div>
      </div>
      <div
        className="line"
      >
        <div
          className="param"
        >
          Coordinates
        </div>
        <div
          className="value"
        >
          [59.86, 30.39]
        </div>
      </div>
    </div>
  </div>
  <div
    id="nav_favour_weather"
  >
    <div
      className="text"
    >
      Favourites
    </div>
    <form
      id="add_favour_form"
      onSubmit={[Function]}
    >
      <input
        className="input_text"
        name="city_name"
        onChange={[Function]}
        placeholder="Add new city"
        required={true}
        type="text"
        value=""
      />
      <input
        alt="add"
        className="img_button"
        name="submit"
        src="img/add.png"
        type="image"
        value="Submit"
      />
    </form>
  </div>
  <div
    id="favour_weather_container"
  >
    <div
      className="favour_block"
    >
      <div
        className="placeholder"
      >
        <div
          className="text"
        >
          Error, try again later..
        </div>
        <img
          alt="loading"
          className="placeholder_img"
          src="img/error.gif"
        />
        <input
          alt="delete"
          className="img_button"
          name="submit"
          onClick={[Function]}
          src="img/del.png"
          type="image"
        />
      </div>
    </div>
    <div
      className="favour_block"
    >
      <div
        className="placeholder"
      >
        <div
          className="text"
        >
          Wait, weather loading is in progress..
        </div>
        <img
          alt="loading"
          className="placeholder_img"
          src="img/loading.gif"
        />
      </div>
    </div>
    <div
      className="favour_block"
    >
      <div
        className="main_favour_weather_info"
      >
        <div
          className="city"
        >
          Paris
        </div>
        <div
          className="img"
        >
          <img
            alt="weather"
            className="weather_icon"
            src="http://openweathermap.org/img/wn/10n@2x.png"
          />
        </div>
        <div
          className="temp"
        >
          5
          °C
        </div>
        <input
          alt="delete"
          className="img_button"
          name="submit"
          onClick={[Function]}
          src="img/del.png"
          type="image"
        />
      </div>
      <div
        className="list"
        id="advanced_weather_info"
      >
        <div
          className="line"
        >
          <div
            className="param"
          >
            Weather
          </div>
          <div
            className="value"
          >
            Light rain
          </div>
        </div>
        <div
          className="line"
        >
          <div
            className="param"
          >
            Pressure
          </div>
          <div
            className="value"
          >
            753 mmHg
          </div>
        </div>
        <div
          className="line"
        >
          <div
            className="param"
          >
            Humidity
          </div>
          <div
            className="value"
          >
            93 %
          </div>
        </div>
        <div
          className="line"
        >
          <div
            className="param"
          >
            Wind
          </div>
          <div
            className="value"
          >
            2.1
          </div>
        </div>
        <div
          className="line"
        >
          <div
            className="param"
          >
            Coordinates
          </div>
          <div
            className="value"
          >
            [48.86, 2.35]
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
